pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/HiveDataFragmenter.java

```diff
@@ -405,24 +411,41 @@ private String buildFilterStringForHive() throws Exception {
         HiveFilterBuilder eval = new HiveFilterBuilder(inputData);
         Object filter = eval.getFilterObject(filterInput);
 
-        String prefix = "";
-
-        if (filter instanceof List) {
-
-            for (Object f : (List<?>) filter) {
-                if (buildSingleFilter(f, filtersString, prefix)) {
-                    // Set 'and' operator between each matched partition filter.
-                    prefix = " and ";
-                }
-            }
-
+        if (filter instanceof LogicalFilter) {
+            buildCompoundFilter((LogicalFilter) filter, filtersString);
         } else {
-            buildSingleFilter(filter, filtersString, prefix);
+            buildSingleFilter(filter, filtersString, "");
         }
 
         return filtersString.toString();
     }
 
+    private void buildCompoundFilter(LogicalFilter filter, StringBuilder filterString) throws Exception{
+        String prefix;
+        switch(filter.getOperator()) {
+            case HDOP_AND:
+                prefix = " and ";
+                break;
+            case HDOP_OR:
+                prefix = " or ";
+                break;
+            case HDOP_NOT:
+                prefix = " not ";
+                break;
+            default:
+                prefix = "";
+        }
+
+        for (int i = 0; i < filter.getFilterList().size(); i++) {
```

Comment:
for (Object f : filter.getFilterList() )

is more compact notation that collapses 2 lines into one